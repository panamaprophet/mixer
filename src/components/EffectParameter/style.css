.parameter {
    position: relative;

    display: flex;

    margin: 8px 8px 16px;
}

.controlContainer {
    flex-grow: 1;

    width: 40%;
    margin-right: 16px;
}

.title {
    font-weight: bold;

    display: inline-block;
    overflow: hidden;

    width: 35%;
    margin-right: 5%;
    margin-left: 8px;

    text-align: left;
    white-space: nowrap;
    text-overflow: ellipsis;

    color: #555;
    text-shadow: 1px 1px 0 #eee;
}
