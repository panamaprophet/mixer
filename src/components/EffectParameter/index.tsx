import React from 'react';

import Fader from '/components/Fader';
import Radio from '/components/Radio';

import style from './style.css';


const EffectParameterMap = {
    'fader': Fader,
    'radio': Radio,
};

const getControlByType = type => EffectParameterMap[type];


const EffectParameter = ({
    onChange,
    ...props
}) => {
    const {
        id,
        name,
        type = 'fader',
    } = props;

    const Control = getControlByType(type);

    return Control
    ? (
        <div className={style.parameter} key={id}>
            <span className={style.title}>{name}:</span>
            <div className={style.controlContainer}>
                <Control {...props} onChange={onChange(id)} />
            </div>
        </div>
    ) : null;
};

export default EffectParameter;