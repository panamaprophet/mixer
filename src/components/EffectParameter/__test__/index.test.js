import React from 'react';
import {shallow} from 'enzyme';

import EffectParameter from '../';


const propsMock = {
    id: 'delay',
    name: 'Delay',
    type: 'radio',
    onChange: jest.fn(),
};

describe('<EffectParameter />', () => {
	it('should renders without any errors', () => {
		const wrapper = shallow(<EffectParameter {...propsMock} />);

		expect(wrapper).toMatchSnapshot();
	});

	describe('type', () => {
		it('"radio" renders <Radio />', () => {
			const props = {
				...propsMock,
				type: 'radio',
			};

			const wrapper = shallow(<EffectParameter {...props} />);
			const result = wrapper.find('.controlContainer').find('Radio').length === 1;

			expect(result).toBe(true);
		});

		it('"fader" renders <Fader />', () => {
			const props = {
				...propsMock,
				type: 'fader',
			};

			const wrapper = shallow(<EffectParameter {...props} />);
			const result = wrapper.find('.controlContainer').find('Fader').length === 1;

			expect(result).toBe(true);
		});


		it('returns nothing for unsupported type', () => {
			const props = {
				...propsMock,
				type: 'weed',
			};

			const wrapper = shallow(<EffectParameter {...props} />);
			const result = wrapper.isEmptyRender();

			expect(result).toBe(true);
		});
	});

	describe('onChange', () => {
		it('should be called on control change', () => {
			const wrapper = shallow(<EffectParameter {...propsMock} />);

			wrapper.find('Radio').at(0).simulate('click');

			expect(propsMock.onChange).toBeCalled();
		});
	});
});