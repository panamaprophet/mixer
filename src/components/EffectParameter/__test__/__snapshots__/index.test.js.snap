// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EffectParameter /> should renders without any errors 1`] = `
<div
  className="parameter"
  key="delay"
>
  <span
    className="title"
  >
    Delay
    :
  </span>
  <div
    className="controlContainer"
  >
    <Radio
      id="delay"
      name="Delay"
      type="radio"
    />
  </div>
</div>
`;
